From 915bc91b6f21d41c0e57f354605c4629acc816e9 Mon Sep 17 00:00:00 2001
From: Pramod Kumar Yadav <pramodx.k.yadav@intel.com>
Date: Thu, 22 Sep 2016 17:30:24 +0530
Subject: [PATCH 4141/4301] ASoC: Intel: Skylake: get dma buf size for
 modules.

commit f3674572fe7931a0febd0fb3b33249f0292f4dde from
git://git@github.com:01org/linux-intel-4.9.git

DMA buffer size for a module is now can be fetched from topology
binary. This buffer size is applicable for gateway copier
modules.

Change-Id: I8ff57726f48e58c2e23b2865eec28abc9ab8d14c
Signed-off-by: Pramod Kumar Yadav <pramodx.k.yadav@intel.com>
Signed-off-by: Ramesh Babu <ramesh.babu@intel.com>
Signed-off-by: Subhransu S. Prusty <subhransu.s.prusty@intel.com>
Signed-off-by: Mallikarjun, chippalkatti <mallikarjunx.chippalkatti@intel.com>
Reviewed-on: https://git-gar-1.devtools.intel.com/gerrit/15040
Reviewed-by: audio_build <audio_build@intel.com>
Reviewed-by: Sm, Bhadur A <bhadur.a.sm@intel.com>
Tested-by: Sm, Bhadur A <bhadur.a.sm@intel.com>
---
 include/uapi/sound/snd_sst_tokens.h    |    6 ++++--
 sound/soc/intel/skylake/skl-topology.c |    4 ++++
 sound/soc/intel/skylake/skl-topology.h |    1 +
 3 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/include/uapi/sound/snd_sst_tokens.h b/include/uapi/sound/snd_sst_tokens.h
index 2eea4da..a5a4242 100644
--- a/include/uapi/sound/snd_sst_tokens.h
+++ b/include/uapi/sound/snd_sst_tokens.h
@@ -152,7 +152,7 @@
  * %SKL_TKN_U32_CAPS_SIZE:      Caps size
  *
  * %SKL_TKN_U32_PROC_DOMAIN:    Specify processing domain
- *
+
  * %SKL_TKN_U32_LIB_COUNT:      Specifies the number of libraries
  *
  * %SKL_TKN_STR_LIB_NAME:       Specifies the library name
@@ -317,6 +317,7 @@
  * %SKL_TKN_U32_AGG_ID:         Aggregation id is a non zero identifier to
  *                              indicate if this endpoint is participating
  *                              in aggregation.
+ * %SKL_TKN_U32_DMA_BUF_SIZE:	DMA buffer size in millisec
  *
  * module_id and loadable flags dont have tokens as these values will be
  * read from the DSP FW manifest
@@ -461,7 +462,8 @@ enum SKL_TKNS {
 	SKL_TKN_U32_AGG_CH_MASK,
 	SKL_TKN_U32_AGG_ID,
 
-	SKL_TKN_MAX = SKL_TKN_U32_AGG_ID,
+	SKL_TKN_U32_DMA_BUF_SIZE,
+	SKL_TKN_MAX = SKL_TKN_U32_DMA_BUF_SIZE,
 };
 
 #endif
diff --git a/sound/soc/intel/skylake/skl-topology.c b/sound/soc/intel/skylake/skl-topology.c
index c08f713..11a04a0 100644
--- a/sound/soc/intel/skylake/skl-topology.c
+++ b/sound/soc/intel/skylake/skl-topology.c
@@ -2978,6 +2978,10 @@ static int skl_tplg_get_token(struct device *dev,
 				tkn_elem->value;
 		break;
 
+	case SKL_TKN_U32_DMA_BUF_SIZE:
+		mconfig->dma_buffer_size = tkn_elem->value;
+		break;
+
 	case SKL_TKN_U8_IN_PIN_TYPE:
 	case SKL_TKN_U8_OUT_PIN_TYPE:
 	case SKL_TKN_U8_CONN_TYPE:
diff --git a/sound/soc/intel/skylake/skl-topology.h b/sound/soc/intel/skylake/skl-topology.h
index 5fe9e60..b57bc3c 100644
--- a/sound/soc/intel/skylake/skl-topology.h
+++ b/sound/soc/intel/skylake/skl-topology.h
@@ -436,6 +436,7 @@ struct skl_module_cfg {
 	u32 sdw_stream_num;
 	bool sdw_agg_enable;
 	struct skl_sdw_aggregation sdw_agg;
+	u32 dma_buffer_size; /* in milli seconds */
 	struct skl_module_pin *m_in_pin;
 	struct skl_module_pin *m_out_pin;
 	enum skl_module_type m_type;
-- 
1.7.5.4

